agent/engine/docker_task_engine.go

```diff
@@ -794,6 +794,9 @@ func (engine *DockerTaskEngine) AddTask(task *apitask.Task) {
 		// This will update the container desired status
 		task.UpdateDesiredStatus()
 
+		// This will update any dependencies for awsvpc network mode before the task is started.
+		engine.updateTaskENIDependencies(task)
```

Comment:
Feels like this needs to be invoked in `PostUnmarshalTask` since that's where most of the task setup happens.

Reply:
These dependencies should be fulfilled only once. `PostUnmarshalTask` is called for each task payload received by the agent which involves task updates.
Therefore, we are calling this method just before adding the task to the engine.
We have validated that the agent checkpoints all the relevant fields and has updated information on restart.